File: Elite2021/armstrong_number.hs
-- digits :: Int -> [Int] --
digits 0 = []
digits n = digits (quot n 10) ++ [(rem n 10)]

powers :: [Int] -> Int -> [Int]
powers d n = map (^n) d

isArmstrong :: Int -> Bool
isArmstrong n = n == sum (powers d (length d)) where d = digits n)

genArmstrong :: Int -> Int -> [Int]
genArmstrong n n' = filter isArmstrong [n..n']

main = do 
    print $ genArmstrong 100 1000
    print $ genArmstrong 1000 10000

File: armstrongNumber.py
def is_armstrong(n: int) -> bool:
    digits = [int(x) for x in str(n)]
    return n == sum([x ** len(digits) for x in digits])

def make_armstrong(start: int, limit: int) -> [int]:
    return [x for x in range(start, limit) if is_armstrong(x)]

def isArmstrong(n: int) -> bool:
    digits = list(map(int, str(n))) 
    return n == sum(map(lambda r: r ** len(digits), digits))

def makeArmstrong(start: int, limit: int) -> [int]:
    return filter(is_armstrong, range(start, limit))

print(make_armstrong(100, 100000))
print(list(makeArmstrong(100, 100000))


